src/SFA.DAS.Tools.Servicebus.Support.Infrastructure/Services/CosmosDbContext.cs

```diff
@@ -93,6 +106,26 @@ public async Task<IEnumerable<QueueMessage>> GetQueueMessagesAsync(string userId
             return messages;                          
         }
 
+        public async Task<IEnumerable<QueueMessage>> GetQueueMessagesByIdAsync(string userId, IEnumerable<string> ids)
+        {
+            var idsList = "\"" + string.Join($"\",\"", ids.Select(x => x.ToString()).ToArray()) + "\"";
+            var sqlQuery = $"SELECT * FROM c WHERE c.userId ='{userId}' and c.id in ({idsList})";
```

Comment:
Can we start to look at using the [Linq to SQl](https://docs.microsoft.com/en-us/azure/cosmos-db/sql-query-linq-to-sql) api?